src/spatio_temporal_voxel_grid.cpp

```diff
@@ -204,6 +207,10 @@ void SpatioTemporalVoxelGrid::TemporalClearAndGenerateCostmap(                \
           {
             std::cout << "Failed to clear point." << std::endl;
           }
+          else
+          {
+            updated_cells.insert(occupany_cell(pose_world[0], pose_world[1]));
```

Comment:
This could go before, not in, the `if` statement, the `if` is only for logging errors in multithreading. Same with L240 below.

Reply:
I did it like this because if the clearing failed, I don't see a reason to touch those cells - after all, they didn't change.

Do you mean that when multithreading `SpatioTemporalVoxelGrid::ClearGridPoint` might return false even though the clearing succeeded? If that's the case, then I agree that this shouldn't be inside the `else` statement.

Reply:
> I did it like this because if the clearing failed, I don't see a reason to touch those cells - after all, they didn't change.

That's a totally fair assumption, but honestly I should remove that logging, it never triggers, it was more for development while there were situations where it was being called. Also, if we wanted to remove the point, we should also touch it for update in the costmap. These points are very likely grouped together so if 1 point is missed, its not like it will impact the bounds very much but we don't want to be in a situation where its problematic. 

Since this is just listing cells to touch for update -- and not actually updating their values directly -- we don't need to be as careful about it to have the `else` since it doesn't represent a direct update, just a request to _be_ updated

Reply:
Fixed